core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -375,14 +383,21 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
         topicPartition -> partitionState
       }
 
-      val version = if (controller.config.interBrokerProtocolVersion >= KAFKA_0_10_0_IV1) 2: Short
+      val version :Short  = if (controller.config.interBrokerProtocolVersion >= KAFKA_0_10_0_IV1) 2: Short
                     else if (controller.config.interBrokerProtocolVersion >= KAFKA_0_9_0) 1: Short
                     else 0: Short
 
       val updateMetadataRequest =
         if (version == 0) {
-          val liveBrokers = controllerContext.liveOrShuttingDownBrokers.map(_.getNode(SecurityProtocol.PLAINTEXT))
-          new UpdateMetadataRequest(controllerId, controllerEpoch, liveBrokers.asJava, partitionStates.asJava)
+          // Version 0 of UpdateMetadataRequest only supports PLAINTEXT.
+          val liveBrokers = controllerContext.liveOrShuttingDownBrokers.map { broker =>
+            val node = broker.getNode(SecurityProtocol.PLAINTEXT)
+            val endPoints = Map{ SecurityProtocol.PLAINTEXT -> new EndPoint(node.host(), node.port()) }
```

Comment:
Nit: the map creation is usually done like `Map(SecurityProtocol.PLAINTEXT -> new EndPoint(node.host, node.port))`.